lib/services/android-plugin-build-service.ts

```diff
@@ -172,23 +173,79 @@ export class AndroidPluginBuildService implements IAndroidPluginBuildService {
 		this.validateOptions(options);
 		const manifestFilePath = this.getManifest(options.platformsAndroidDirPath);
 		const androidSourceDirectories = this.getAndroidSourceDirectories(options.platformsAndroidDirPath);
-		const shouldBuildAar = !!manifestFilePath || androidSourceDirectories.length > 0;
+		const shortPluginName = getShortPluginName(options.pluginName);
+		const pluginTempDir = path.join(options.tempPluginDirPath, shortPluginName);
+		const pluginSourceFileHashesInfo = await this.getSourceFilesHashes(options.platformsAndroidDirPath, shortPluginName);
+
+		const shouldBuildAar = await this.shouldBuildAar({
+			manifestFilePath,
+			androidSourceDirectories,
+			pluginTempDir,
+			pluginSourceDir: options.platformsAndroidDirPath,
+			shortPluginName,
+			fileHashesInfo: pluginSourceFileHashesInfo
+		});
 
 		if (shouldBuildAar) {
-			const shortPluginName = getShortPluginName(options.pluginName);
-			const pluginTempDir = path.join(options.tempPluginDirPath, shortPluginName);
-			const pluginTempMainSrcDir = path.join(pluginTempDir, "src", "main");
+			// In case plugin was already built in the current process, we need to clean the old sources as they may break the new build.
```

Comment:
move these 3 lines in something like this.clean(pluginTempDir) or even fs.clean in order to keep the method as short and readable as possible